packages/core/src/scenes/GeneratorScene.ts

```diff
@@ -263,6 +266,7 @@ export abstract class GeneratorScene<T>
     } else {
       this.state = SceneState.Initial;
     }
+    this.variables.destroySignals();
```

Comment:
Similarly to `TimeEvents`, `Variables` should subscribe to scene events themselves.
In the constructor of `Varialbes` you should do:
```ts
scene.onReset.subscribe(this.handleReset);
```
`GeneratorScene` is only the default implementation of the `Scene` interface. There will be more in the future. Doing registration in the `Variables` class makes it easier to reuse them in different scene implementations.